src/NuGet.Core/NuGet.Configuration/Settings/Items/CredentialsItem.cs

```diff
@@ -149,12 +140,13 @@ public CredentialsItem(string name, string username, string password, bool isPas
         internal CredentialsItem(XElement element, SettingsFile origin)
             : base(element, origin)
         {
-            ElementName = element.Name.LocalName;
+            // ElementName is read from XML file, so it must be decoded.
+            ElementName = XmlConvert.DecodeName(element.Name.LocalName);
 
             var elementDescendants = element.Elements();
             var countOfDescendants = elementDescendants.Count();
 
-            var parsedItems = elementDescendants.Select(e => SettingFactory.Parse(e, origin) as AddItem).Where(i => i != null);
+            var parsedItems = elementDescendants.Select(e => (AddItem)SettingFactory.Parse(e, origin));
```

Comment:
Before the logic was using a safe cast (`as AddItem`) which would return `null` if the item was not of that type.  It would not throw an exception right?  Then it would check for `null` and finally cast the item.  `.OfType<T>()` does all of that in a single call and is used in other places in this assembly.

Here's a sample of the three implementations: https://dotnetfiddle.net/w8kV4G

```suggestion
            var parsedItems = elementDescendants.Select(e => SettingFactory.Parse(e, origin)).OfType<AddItem>();
```

Reply:
You're right. Fixed in the next commit.